### PR TITLE
[bug] Fix fanet issue for negative lat/long

### PR DESCRIPTION
**Pull request by kolupaev**
FANET protocol uses 2 compliment 24bit integers.

As C does not support it, adding a manual conversion.

Issue reproducible in north-america, and manifests as glider showing far left.

After the fix location is displayed correctly. Verified via soft-rf transmitter.

See https://github.com/fhorinek/BB/issues/497

### CODE REVIEW

File: BB3/App/drivers/gnss/fanet.c [hunk 1]
```diff
@@ -285,30 +285,35 @@ void fanet_parse_dg(char * buffer)
...
+	tmp.u8[2] = data[2];
+	*lat = (complement2_24bit(tmp.u32) / 93206.0) * GNSS_MUL;
+
+	//longitude
+	//memcpy((void *)&tmp, data + 3, 3);
+	tmp.u32 = 0;
+	tmp.u8[0] = data[3];
+	tmp.u8[1] = data[4];
+	tmp.u8[2] = data[5];
+	*lon = (complement2_24bit(tmp.u32) / 46603.0) * GNSS_MUL;
```
Comment (bubeck) [inline BB3/App/drivers/gnss/fanet.c:304]: What is 46603 or 93206? This should probably be a #define?

**Comment by bubeck** [comment]
In general this looks good to me. However, I am unable to test, because I live in germany (positive lat/lon).

File: BB3/App/drivers/gnss/fanet.c (hunk 1 above)
Comment (kolupaev) [reply BB3/App/drivers/gnss/fanet.c:304]: Those are constants specific to fanet conversion. 
Sure thing, will extract into defines / constants.

**Comment by kolupaev** [comment]
> In general this looks good to me. However, I am unable to test, because I live in germany (positive lat/lon).

Tested it locally with Soft-rf, works as expected on negative. Would that be sufficient?

**Comment by bubeck** [comment]
> Tested it locally with Soft-rf, works as expected on negative. Would that be sufficient?

Must be decided by @fhorinek

**Comment by kolupaev** [comment]
Ah, Thanks for CC-ing right people :) 
Was not sure who is a project owner here

**Pull request merged**